### ISSUE TITLE
[CIT] Legacy NBT paths falsely matching as valid in 1.21.1 pre9

### ISSUE THREAD

**Issue by SeaOfPixels** [sp614x/optifine#7742]
## Description of Issue
CIT that is not based on renames (like unique potions using CIT) now use components instead of NBT. While this is correct, the NBT parameter is still being incorrectly recognised and applied in a glitchy way, when it shouldn't be recognised at all. This is a problem when trying to make a pack work for 1.20.4 _and_ 1.20.6/1.21/1.21.1, because in this case you need both parameters (components AND nbt) to be present. The NBT parameter needs to be ignored entirely in 1.20.6/1.21/1.21.1 for non-rename based CIT.

## Steps to Reproduce
1. Open Minecraft
2. Enable resource pack in the resource packs menu that contains component-based potion CIT _and_ nbt-based potion CIT
3. Notice how the nbt parameter is causing issues even when having components take priority

## OptiFine Version
OptiFine 1.21.1_HD_U_J1_pre9

## Screenshots
Notice how the nbt is causing Water Bottles/Mundane Potions/Thick Potions/Awkward Potions to be given the Fire Resistance potion texture when nbt lines shouldn't be getting read in 1.21.1 in the first place.
![2024-09-04_18 32 44](https://github.com/user-attachments/assets/1d57fc1d-5b1e-43df-9187-5d1ce2348d09)

## Parameters comparison
Components for 1.20.6/1.21/1.21.1 (recognised):
```
type=item
items=potion
model=item/fire_resistance_potion
components.potion_contents.potion=minecraft:fire_resistance
```

NBT for 1.20.4 (should not be recognised in 1.20.6/1.21/1.21.1, because it is applying in a glitchy way):
```
type=item
items=potion
model=item/fire_resistance_potion
nbt.Potion=minecraft:fire_resistance
```

**Comment by ZenithRogue** [comment]
Old NBT paths _should_ be ignored, and flag as invalid, definitely shouldnt be matching like this

**Comment by JimStoneCraft** [comment]
The nbt line (nbt.Potion=minecraft:fire_resistance) is being ignored in 1.12.1, what the properties file is asking for is fire_resistance_potion texture for all potions.

**Comment by ewanhowell5195** [comment]
this is working as intended. the nbt line is ignored in 1,21, and since all provided rules are true (you have no rules), the texture is applied.

providing
```properties
type=item
items=potion
model=item/fire_resistance_potion
nbt.Potion=minecraft:fire_resistance
```
in 1.21 is the same as providing
```providing
type=item
items=potion
model=item/fire_resistance_potion
```
and since it has no requirements, it gets applied

This behaviour is a good thing, as it allows for backwards compatibility. if you want it to work in both 1.20 and 1.21, you can provide
```properties
type=item
items=potion
model=item/fire_resistance_potion
components.potion_contents.potion=minecraft:fire_resistance
nbt.Potion=minecraft:fire_resistance
```
although you shouldnt do this, you should use overlay packs

**Comment by ZenithRogue** [comment]
As `nbt.` is still a valid property in 1.21 CIT (for of `nbt.display.Name`/`nbt.display.Lore` ), `nbt.Potion` should be an invalid match.

SHsuperCM, the CITresewn dev said the following for their implementation:
> I made the cit parsing fail if there is nbt (that is not name or lore) in 1.21, it just skips parsing the file and logs which one has a problem and what pack it's from
I'd rather it be a logged error I could point out in issues

**Comment by sp614x** [comment]
Unrecognized NBT rules should only cause the properties file to be skipped if no components are specified, as then it will always match:
```
type=item
items=potion
model=item/fire_resistance_potion
nbt.Potion=minecraft:fire_resistance
```
When both `nbt` and `components` are present the unrecognized `nbt` is ignored (only warning) as the `components` are used to match correctly:
```
type=item
items=potion
model=item/fire_resistance_potion
components.potion_contents.potion=minecraft:fire_resistance
nbt.Potion=minecraft:fire_resistance
```
This will allow the RP to work in both old and new versions.

**Comment by ZenithRogue** [comment]
> This will allow the RP to work in both old and new versions.
IMO if pack developers want to address compatibility, they should be using the new pack overlays instead of mix and matching invalid components, which could create other issues

**Comment by ZenithRogue** [comment]
![image](https://github.com/user-attachments/assets/57566fd1-2484-427a-ae12-ff86176c502c)
fixed in OptiFine 1.21.1_HD_U_J1_pre11
